framework/src/media/StreamHandler.cpp

```diff
@@ -0,0 +1,190 @@
+/* ****************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include "StreamHandler.h"
+
+#include <debug.h>
+
+#ifndef CONFIG_HANDLER_STREAM_BUFFER_SIZE
+#define CONFIG_HANDLER_STREAM_BUFFER_SIZE 4096
+#endif
+
+#ifndef CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD
+#define CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD 2048
+#endif
+
+namespace media {
+namespace stream {
+
+StreamHandler::StreamHandler() :
+	mWorker(0),
+	mWorkerName("StreamWorker"),
+	mWorkerStackSize(4096),
+	mIsWorkerAlive(false)
+{
+}
+
+bool StreamHandler::open()
+{
+	if (!getStreamBuffer()) {
+		auto streamBuffer = StreamBuffer::Builder()
+								.setBufferSize(CONFIG_HANDLER_STREAM_BUFFER_SIZE)
+								.setThreshold(CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD)
+								.build();
+
+		if (!streamBuffer) {
+			meddbg("streamBuffer is nullptr!\n");
+			return false;
+		}
+
+		setStreamBuffer(streamBuffer);
+	}
+
+	if (!(mDataSource->open() && registerCodec(mDataSource->getAudioType(), mDataSource->getChannels(), mDataSource->getSampleRate()))) {
+		return false;
+	}
+
+	return start();
+}
+
+bool StreamHandler::close()
+{
+	stop();
+	unregisterCodec();
+	return mDataSource->close();
+}
+
+bool StreamHandler::start()
+{
+	medvdbg("StreamHandler::start()\n");
+	if (mDataSource && !mDataSource->isPrepared()) {
+		return false;
+	}
+
+	createWorker();
+	return mIsWorkerAlive;
+}
+
+bool StreamHandler::stop()
+{
+	destroyWorker();
+	return true;
+}
+
+void StreamHandler::createWorker()
+{
+	medvdbg("StreamHandler::createWorker()\n");
+	if (mStreamBuffer && !mIsWorkerAlive) {
+		resetWorker();
```

Comment:
resetWorker -> resetStreamWorker
then you can do below things inside of resetStreamWorker;
```
mStreamBuffer = nullptr;
mBufferWriter->setEndOfStream();
...
```
please think about another function, too.
Not this PR actually.

Reply:
I'll raise another PR